File: api/BeSureApi/Models/BankBranch.cs
using System.ComponentModel.DataAnnotations;

namespace BeSureApi.Models
{
    public class BankBranchDetails
    {
        public int Id { get; set; }
        public int BankId { get; set; }
        public string BankName { get; set; }
        public string BranchCode { get; set; }
        public int BranchId { get; set; }
        public string BranchName { get; set; }
        public string Address { get; set; }
        public int CityName { get; set; }
        public int StateName { get; set; }
        public int CountryId { get; set; }
        public string Pincode { get; set; }
        public string ContactPerson { get; set; }
        public string ContactNumberOneCountryCode { get; set; }
        public string ContactNumberOne { get; set; }
        public string? ContactNumberTwoCountryCode { get; set; }
        public string? ContactNumberTwo { get; set; }
        public string Email { get; set; }
        public string Ifsc { get; set; }
        public string MicrCode { get; set; }
        public string SwiftCode { get; set; }
        public DateTime EffectiveFrom { get; set; }
        public DateTime? EffectiveTo { get; set; }
        public string CreatedByFullName { get; set; }
        public DateTime CreatedOn { get; set; }
        public string? UpdatedByFullName { get; set; }
        public DateTime? UpdatedOn { get; set; }
        public bool IsDeleted { get; set; }
        public int? DeletedBy { get; set; }
        public DateTime? DeletedOn { get; set; }
    }
    public class BankBranchList
    {
        public int Id { get; set; }
        public string BankName { get; set; }
        public string BranchName { get; set; }
        public string CityName { get; set; }
        public string StateName { get; set; }
        public string Ifsc { get; set; }
        public string CreatedByFullName { get; set; }
        public DateTime CreatedOn { get; set; }
    }
    public class BankBranchCreate
    {
        [Required(ErrorMessage = "validation_error_bankbranch_create_bank_required")]
        [Range(1, int.MaxValue, ErrorMessage = "validation_error_bankbranch_create_bank_required")]
        public int BankId { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_code_required")]
        public string BranchCode { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_name_required")]
        public string BranchName { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_address_required")]
        public string Address { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_city_required")]
        [Range(1, int.MaxValue, ErrorMessage = "validation_error_bankbranch_create_city_required")]
        public int CityId { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_state_required")]
        [Range(1, int.MaxValue, ErrorMessage = "validation_error_bankbranch_create_state_required")]
        public int StateId { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_country_required")]
        [Range(1, int.MaxValue, ErrorMessage = "validation_error_bankbranch_create_country_required")]
        public int CountryId { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_pincode_required")]
        public string Pincode { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_contactperson_required")]
        public string ContactPerson { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_contact1countrycode_required")]
        public string ContactNumberOneCountryCode { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_contact1_required")]
        public string ContactNumberOne { get; set; }
        public string? ContactNumberTwoCountryCode { get; set; }
        public string? ContactNumberTwo { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_email_required")]
        public string Email { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_ifsc_required")]
        public string Ifsc { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_micrcode_required")]
        public string MicrCode { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_swiftcode_required")]
        public string SwiftCode { get; set; }
        public int CreatedBy { get; set; }
    }

    public class BankBranchEdit
    {
        public int Id { get; set; }
        public int BranchId { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_code_required")]
        public string BranchCode { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_name_required")]
        public string BranchName { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_address_required")]
        public string Address { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_city_required")]
        [Range(1, int.MaxValue, ErrorMessage = "validation_error_bankbranch_create_city_required")]
        public int CityId { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_state_required")]
        [Range(1, int.MaxValue, ErrorMessage = "validation_error_bankbranch_create_state_required")]
        public int StateId { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_country_required")]
        [Range(1, int.MaxValue, ErrorMessage = "validation_error_bankbranch_create_country_required")]
        public int CountryId { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_pincode_required")]
        public string Pincode { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_contactperson_required")]
        public string ContactPerson { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_contact1countrycode_required")]
        public string ContactNumberOneCountryCode { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_contact1_required")]
        public string ContactNumberOne { get; set; }
        public string? ContactNumberTwoCountryCode { get; set; }
        public string? ContactNumberTwo { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_email_required")]
        public string Email { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_ifsc_required")]
        public string Ifsc { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_micrcode_required")]
        public string MicrCode { get; set; }
        [Required(ErrorMessage = "validation_error_bankbranch_create_swiftcode_required")]
        public string SwiftCode { get; set; }
    }
}

File: ui/src/types/bankBranch.ts
import { Decoder, array, boolean, nullable, number, object, string } from "decoders";

export interface BankBranchCreate {
  BankId: number | string;
  BranchCode: string;
  BranchName: string;
  Address: string;
  CityId: number | null;
  StateId: number | null;
  CountryId: number | null;
  Pincode: string;
  ContactPerson: string;
  ContactNumberOneCountryCode: string;
  ContactNumberOne: string;
  ContactNumberTwoCountryCode: string;
  ContactNumberTwo: string;
  Email: string;
  Ifsc: string;
  MicrCode: string;
  SwiftCode: string;
}

export interface BankBranchCreateResult {
  IsBankBranchCreated: Boolean;
}

export const createBankBranchDecoder: Decoder<BankBranchCreateResult> = object({
  IsBankBranchCreated: boolean,
});

export interface BankBranchUpdateResult {
  IsBankBranchUpdated: Boolean;
}

export const createBankBranchUpdateDecoder: Decoder<BankBranchUpdateResult> = object({
  IsBankBranchUpdated: boolean,
});

export interface BankBranchListDetails {
  Id: number,
  BankName: string;
  BranchName: string;
  CityName: string;
  StateName: string;
  Ifsc: string;
  CreatedByFullName: string;
  CreatedOn: string;
}

export const bankBranchListDetailsDecoder: Decoder<BankBranchListDetails> = object({
  Id: number,
  BankName: string,
  BranchName: string,
  CityName: string,
  StateName: string,
  Ifsc: string,
  CreatedByFullName: string,
  CreatedOn: string,
});

export interface BankBranchList {
  BankBranches: BankBranchListDetails[],
  TotalRows: number;
  PerPage: number;
}

export const bankBranchListDecoder: Decoder<BankBranchList> = object({
  BankBranches: array(bankBranchListDetailsDecoder),
  TotalRows: number,
  PerPage: number
});

export interface BankBranchDetails {
  Id: number;
  BankId: number;
  BankName: string;
  BranchCode: string;
  BranchName: string;
  Address: string;
  CityName: number;
  StateName: number;
  CountryId: number;
  Pincode: string;
  ContactPerson: string;
  ContactNumberOneCountryCode: string;
  ContactNumberOne: string;
  ContactNumberTwoCountryCode: string;
  ContactNumberTwo: string;
  Email: string;
  Ifsc: string;
  MicrCode: string;
  SwiftCode: string;
  CreatedByFullName: string;
  CreatedOn: string;
  UpdatedByFullName: null | string;
  UpdatedOn: null | string;
}

export const bankBranchDetailsDecoder: Decoder<BankBranchDetails> = object({
  Id: number,
  BankName: string,
  BankId: number,
  BranchCode: string,
  BranchName: string,
  Address: string,
  CityName: number,
  StateName: number,
  CountryId: number,
  Pincode: string,
  ContactPerson: string,
  ContactNumberOneCountryCode: string,
  ContactNumberOne: string,
  ContactNumberTwoCountryCode: string,
  ContactNumberTwo: string,
  Email: string,
  Ifsc: string,
  MicrCode: string,
  SwiftCode: string,
  CreatedByFullName: string,
  CreatedOn: string,
  UpdatedByFullName: nullable(string),
  UpdatedOn: nullable(string)
});

export interface BankBranchDetailsSelect {
  value: any,
  label: any
}

export interface BankBranchesSelect {
  BankBranches: BankBranchDetailsSelect[];
}

export interface BranchInBankDetail {
  Id: number;
  BranchId: number;
  BranchName: string;
}

export const branchInBankDetailDecoder: Decoder<BranchInBankDetail> = object({
  Id: number,
  BranchId: number,
  BranchName: string,
});

export interface BranchInBankList {
  BankBranches: BranchInBankDetail[];
}

export const branchInBankListDecoder: Decoder<BranchInBankList> = object({
  BankBranches: array(branchInBankDetailDecoder),
});

export interface BankBranchInfo {
  Id: number;
  BankId: number;
  StateId: number;
  CityId: number;
  CountryId: number;
  BankName: string;
  BranchCode: string;
  BranchName: string;
  Address: string;
  City: string;
  State: string;
  Country: string;
  Pincode: string;
  ContactPerson: string;
  ContactNumberOneCountryCode: string;
  ContactNumberOne: string;
  ContactNumberTwoCountryCode: string | null;
  ContactNumberTwo: string | null;
  Email: string;
  Ifsc: string;
  MicrCode: string;
  SwiftCode: string;
  CreatedBy: string;
  CreatedOn: string;
  BranchId:number;
}

export const bankBranchInfoDecoder: Decoder<BankBranchInfo> = object({
  Id: number,
  BankId: number,
  StateId: number,
  CityId: number,
  CountryId: number,
  BankName: string,
  BranchCode: string,
  BranchName: string,
  Address: string,
  City: string,
  State: string,
  Country: string,
  Pincode: string,
  ContactPerson: string,
  ContactNumberOneCountryCode: string,
  ContactNumberOne: string,
  ContactNumberTwoCountryCode: nullable(string),
  ContactNumberTwo: nullable(string),
  Email: string,
  Ifsc: string,
  MicrCode: string,
  SwiftCode: string,
  CreatedBy: string,
  CreatedOn: string,
  BranchId:number
});

export interface SelectedBankBranch {
  BankBranchDetails: BankBranchInfo,
}

export const selectedBankBranchDecoder: Decoder<SelectedBankBranch> = object({
  BankBranchDetails: bankBranchInfoDecoder,
});

export interface BankBranchDeleted {
  IsDeleted: Boolean;
}

export const bankBranchDeletedDecoder: Decoder<BankBranchDeleted> = object({
  IsDeleted: boolean,
});
